lib/ReactViews/Custom/ChartCustomComponent.ts

```diff
@@ -231,22 +231,9 @@ export default abstract class ChartCustomComponent<
 
           return Promise.resolve(itemOrPromise).then(item => {
             if (item) {
+              this.setTraitsFromParent(item, context.catalogItem);
               this.setTraitsFromAttrs(item, attrs, i);
               body && this.setTraitsFromBody?.(item, body);
-              if (
-                hasTraits(
-                  item,
-                  DiscretelyTimeVaryingTraits,
-                  "chartDisclaimer"
-                ) &&
-                chartDisclaimer !== undefined
-              ) {
-                item.setTrait(
-                  CommonStrata.definition,
-                  "chartDisclaimer",
```

Comment:
Apart from some refactoring to move setting of `chartDisclaimer` (and other traits from parent) to a common method `setTraitsFromParent`, we also make sure the trait is set in `CommonStrata.user` so that it gets added to the shared url.